internal-export-file/export-report-pdf/src/export-report-pdf.py

```diff
@@ -509,6 +515,363 @@ def _process_threat_actor_individual(self, entity_id, file_name):
+                    )
+                    continue
+
+                if obj_entity_type not in context["observables"]:
+                    context["observables"][obj_entity_type] = []
+
+                # Defang urls
+                if self.defang_urls and obj_entity_type == "Url":
+                    observable_dict["observable_value"] = observable_dict[
+                        "observable_value"
+                    ].replace("http", "hxxp", 1)
+
+                context["observables"][obj_entity_type].append(observable_dict)
+
+            # Handle all other entities
+            else:
+                reader_func = self._get_reader(obj_entity_type)
+                if reader_func is None:
+                    self.helper.log_error(
+                        f'Could not find a function to read entity with type "{obj_entity_type}"'
+                    )
+                    continue
+                entity_dict = reader_func(id=obj_id)
+
+                if obj_entity_type not in context["entities"]:
+                    context["entities"][obj_entity_type] = []
+
+                context["entities"][obj_entity_type].append(entity_dict)
+
+        # Render html with input variables
+        env = Environment(
+            loader=FileSystemLoader(self.current_dir), finalize=self._finalize
+        )
+        template = env.get_template("resources/case-rfi.html")
+        html_string = template.render(context)
+
+        # Generate pdf from html string
+        pdf_contents = HTML(
+            string=html_string, base_url=f"{self.current_dir}/resources"
+        ).write_pdf()
+
+        # Upload the output pdf
+        self.helper.log_info(f"Uploading: {file_name}")
+        self.helper.api.stix_domain_object.push_entity_export(
+            entity_id, file_name, pdf_contents, "application/pdf"
+        )
+
+    def _process_case_rft(self, entity_id, file_name):
+        """
+        Process a Case Rft entity and upload as pdf.
+        """
+        # Get the Case
+        case_dict = self.helper.api_impersonate.case_rft.read(id=entity_id)
+        content_query = '{case (id:"' + entity_id + '") {content}}'
+        case_dict["content"] = (self.helper.api_impersonate.query(query=content_query))[
+            "data"
+        ]["case"].get("content", "No content available.")
+
+        # Extract values for inclusion in output pdf
+        case_name = case_dict["name"]
+        case_content = case_dict["content"]
+        case_marking = case_dict.get("objectMarking", None)
+        if case_marking:
+            case_marking = case_marking[-1]["definition"]
+        case_external_refs = [
+            external_ref_dict["url"]
+            for external_ref_dict in case_dict["externalReferences"]
+        ]
+        case_confidence = case_dict["confidence"]
+        case_id = case_dict["id"]
+        case_objs = case_dict["objects"]
+        case_date = datetime.datetime.now().strftime("%b %d %Y")
```

Comment:
this is not really the case date, it's the date of the pdf report we are exporting (now). It's just the naming of the variable that could be changed to report_date.

Reply:
indeed